### ISSUE TITLE
Callback not executed when connecting to an active port (not SQL port)

### ISSUE THREAD

**Issue by AHaberl** [SAP/node-hdb#53]
The error occurs with the following coding:
```
var hdb    = require('hdb');
var client = hdb.createClient({
  host     : 'hostname',
  port     : port,
  user     : 'user',
  password : 'secret'
});

client.connect(function (err) {
  if (err) {
    return console.error('Connect error', err);
  }  
});
```
If the port parameter is set to a random unused port, the callback function of client.connect is executed and the error is set. If an active port (e.g. the xs-engine http-port 8000) is used, the callback is never executed and the code exits silently somewhere.  
The expected result would be the execution of the callback with an error like 'wrong port - connection failed' or something like that.

**Comment by petar-iv** [comment]
Hi,

It seems that the connection is closed before calling the callback of the `connect` method. And since the `close` event of the connection is propagated to the client, the `close` event is emitted on the client object.

Check with this code:

```js
var hdb = require('hdb');

var client = hdb.createClient({
  host: 'host',
  port: 'port',
  user: 'user',
  password: 'password'
});

client.on('close', function () {
  console.log('close');
});

client.connect(function (err) {
  if (err) {
    return console.error('Connect error', err);
  }
  console.log('No error');
});
```

The issue is reproducible with earlier versions of the hdb driver as well (like v0.6.0).

Best regards,
Petar Ivanov

**Comment by AHaberl** [comment]
Same experience from my side.
Will there be a fix for this?

**Comment by holgerkoser** [comment]
I can not reproduce the problem. If I run the following script: 

```js
var hdb = require('hdb');

var client = hdb.createClient({
  host: 'localhost',
  port: 2376,
  user: 'user',
  password: 'secret',
  initializationTimeout: 1000
});

client.connect(function (err) {
  if (err) {
    return console.error(err.message);
  }
  console.log('Connected');
});
```
... I will receive the this error after one second.
 
```bash
Could not connect to any host: [ localhost:2376 - No initialization reply received within 1 sec ]
```
On port 2376 an http server is listening, which is not sending the initializaton reply. 

Best, Holger

**Comment by AHaberl** [comment]
Hi, 
@holgerkoser: you are right. The issue does not occur with any http-server. I can only observe it when specifying the http-port of the Hana XS Engine (in my case 8000).

**Comment by xtough** [comment]
any update?

**Comment by alexpenev-s** [comment]
Yes. See https://github.com/SAP/node-hdb/pull/60

**Comment by alexpenev-s** [comment]
Fix should be in v0.9.0
